### PR TITLE
Removed xfail for physics tests and improved stability

### PR DESCRIPTION
**Pull request by AlexOteiza**

### CODE REVIEW

File: AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/utils.py
Comment (moraaar) [inline AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/utils.py:98]: Is it safe to remove this? are other tests relying on this?

File: AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/utils.py
Comment (moraaar) [inline AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/utils.py:45]: After the "wait for condition" is finished probably worth still checking if it's in game mode, and if it is then fail.

File: AutomatedTesting/Gem/PythonTests/Physics/TestSuite_Main_Optimized.py
Comment (moraaar) [inline AutomatedTesting/Gem/PythonTests/Physics/TestSuite_Main_Optimized.py:57]: Remove commented out stuff. Git history on the file can be used to recover it if necessary.

Comment (AMZN-AlexOteiza) [reply AutomatedTesting/Gem/PythonTests/Physics/TestSuite_Main_Optimized.py:78]: Thought I removed, will do

File: AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/utils.py
Comment (AMZN-AlexOteiza) [reply AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/utils.py:151]: No test should ever close the editor. This code is just wrong

File: AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/utils.py
Comment (AMZN-AlexOteiza) [reply AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/utils.py:48]: will do

File: AutomatedTesting/Gem/PythonTests/Physics/TestSuite_Main.py
Comment (AMZN-AlexOteiza) [inline AutomatedTesting/Gem/PythonTests/Physics/TestSuite_Main.py:3]: This was moved TestSuite_Main_Optimized -> TestSuite_Main

Its pretty annoying its not able to tell in the change

**Review by moraaar — DISMISSED** [review]

File: AutomatedTesting/Gem/PythonTests/Physics/TestSuite_Main.py
Comment (amzn-sean) [reply AutomatedTesting/Gem/PythonTests/Physics/TestSuite_Main.py:3]: This is because ```TestSuite_Main.py``` was already in the repo, so it will just appear as changes to the file. Unless you do some extra steps / work (delete old file + commit, then do git mv to rename + commit).

**Review by amzn-sean — DISMISSED** [review]
LGTM

File: AutomatedTesting/Gem/PythonTests/Physics/TestSuite_Main.py
Comment (AMZN-AlexOteiza) [reply AutomatedTesting/Gem/PythonTests/Physics/TestSuite_Main.py:3]: I actually did that if you check the commits, nevermind anyways..

File: AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/utils.py
Comment (greerdv) [inline AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/utils.py:49]: nitpick, typo - should be opening not openning

**Review by greerdv — APPROVED** [review]

**Review by SergeyAMZN — DISMISSED** [review]
The change needs conflicts to be resolved first before merging to development

**Comment by hultonha** [comment]
@AlexOteiza don't suppose there's the faintest chance you'd be able to update and merge this PR? 😅 Otherwise @jckand-amzn is this something you think you could pull in instead? This has all the approvals needed and would be great to get merged if possible 😁 Thanks!

**Comment by AlexOteiza** [comment]
> @AlexOteiza don't suppose there's the faintest chance you'd be able to update and merge this PR? 😅 Otherwise @jckand-amzn is this something you think you could pull in instead? This has all the approvals needed and would be great to get merged if possible 😁 Thanks!

This was held from merging due to some changes that were coming in. I'll see if I can find some time in the weekend :)

**Comment by hultonha** [comment]
> > @AlexOteiza don't suppose there's the faintest chance you'd be able to update and merge this PR? 😅 Otherwise @jckand-amzn is this something you think you could pull in instead? This has all the approvals needed and would be great to get merged if possible 😁 Thanks!
> 
> This was held from merging due to some changes that were coming in. I'll see if I can find some time in the weekend :)

I see, thank you very much for the extra context @AlexOteiza. If you're able to get to this that would be amazing otherwise let us know if @jckand-amzn could take this on. Thanks for getting back to us 🙂

**Comment by lumberyard-employee-dm** [comment]
@moraaar Can you follow up with this PR?

**Comment by moraaar** [comment]
I'll resolve the conflicts and update the PR soon. Thanks.

**Comment by moraaar** [comment]
Resolved conflicts in this PR and run locally.

````
E:\Amazon\GitHub\o3de\AutomatedTesting>..\python\python.cmd -m pytest .\Gem\PythonTests\Physics\TestSuite_Main.py --build-directory=build\windows\bin\profile\
================================================= test session starts =================================================
platform win32 -- Python 3.7.12, pytest-5.3.2, py-1.11.0, pluggy-0.13.1
rootdir: E:\Amazon\GitHub\o3de, inifile: pytest.ini
plugins: mock-2.0.0, timeout-1.3.4, ly-test-tools-1.0.0
collected 86 items

Gem\PythonTests\Physics\TestSuite_Main.py ...xxx.........Xx......x...........................................X.. [ 79%]
.......x..........                                                                                               [100%]

================================================== warnings summary ===================================================
e:\amazon\github\o3de\tools\lytesttools\ly_test_tools\o3de\editor_test.py:333
  e:\amazon\github\o3de\tools\lytesttools\ly_test_tools\o3de\editor_test.py:333: PytestCollectionWarning: cannot collect test class 'TestData' because it has a __init__ constructor (from: AutomatedTesting/Gem/PythonTests/Physics/TestSuite_Main.py::TestAutomation)
    class TestData:

e:\amazon\github\o3de\tools\lytesttools\ly_test_tools\o3de\editor_test.py:333
  e:\amazon\github\o3de\tools\lytesttools\ly_test_tools\o3de\editor_test.py:333: PytestCollectionWarning: cannot collect test class 'TestData' because it has a __init__ constructor (from: AutomatedTesting/Gem/PythonTests/Physics/TestSuite_Main.py::TestAutomationNoPrefab)
    class TestData:

-- Docs: https://docs.pytest.org/en/latest/warnings.html
========================== 80 passed, 6 xfailed, 2 xpassed, 2 warnings in 117.24s (0:01:57) ===========================
````

**Comment by moraaar** [comment]
All the DCO from Alex's commits are failing. I think the best thing to move this forward now would be to create a new branch, do the changes again and do a new PR.
Closing this PR.

**Comment by moraaar** [comment]
Follow up PR with the same changes: https://github.com/o3de/o3de/pull/8066

**Pull request closed without merging**
